### PR TITLE
Update dependencies and test framework

### PR DESCRIPTION
**Pull request by ivacf**
* Update project dependencies to latest version
* Move to robolectric 3.0 and official unit test support 
* Rewrite demo app functional tests in Espresso and remove Robotium.

### CODE REVIEW

File: demo/src/main/java/uk/co/ribot/easyadapterdemo/Person.java [hunk 1]
```diff
@@ -59,4 +59,26 @@ public int getResDrawableId() {
     public void setResDrawableId(int resDrawableId) {
         this.resDrawableId = resDrawableId;
     }
+
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (hitherejoe) [inline demo/src/main/java/uk/co/ribot/easyadapterdemo/Person.java]: really like this parameter naming

**Comment by hitherejoe** [comment]
This looks good. Whats happening with the tests?

**Comment by ivacf** [comment]
Something is wrong with travis and it seems to be killing the process. The test passed the first time so it' safe to merge.

**Comment by hitherejoe** [comment]
If you say so! heart

**Pull request merged**